### ISSUE TITLE
[Core/Spell] Hand of Reckoning

### ISSUE THREAD

**Issue by click** [TrinityCore/TrinityCore#1315]
Core Revision: 11709+ Release Hash: 6ad2278ff59c+ (Win64).
DB: TDB 335.11.39

Hand of Reckoning is suppose to damage fleeing creatures according to retail's data. 

Currently: Hand of Reckoning will ONLY damage fleeing creatures when the creature flees to another creature and calls for help.

What it should do: It should damage ALL fleeing creatures, regardless of whether the creature reaches another creature for help.

Here's my fix I made for you guy's :). I hope it fits your coding standards...
##

**Comment by TULOA** [comment]
Anyone managed to look into this? It is still present in the latest build.

**Comment by tkrokli** [comment]
Cataclysm Patch 4.0.1 (2010-10-12): No longer deals damage.
- https://wow.gamepedia.com/Hand_of_Reckoning
- http://www.wowhead.com/spell=62124/hand-of-reckoning#comments:id=1144229

**Comment by TULOA** [comment]
Yes but this is a wrath server core where its still present. It should be using the Wrath version.

**Comment by tkrokli** [comment]
My comment was meant to explain why I added the label [Branch-3.3.5a] and also to remind those who might have forgotten it, but I guess I should have written that too in the comment as an explanation of why I wrote that comment.

**Comment by TULOA** [comment]
Ahh my mistake.

I did notice the one patch for this regarding the pets scenario helped a little but still is acting up.

Patch I am referencing is here:
https://github.com/TrinityCore/TrinityCore/issues/9068

**Comment by TerrorBringer** [comment]
In rev https://github.com/TrinityCore/TrinityCore/commit/8f91a85aaa789276e0b9a6486ae96b978ec281d9 Hand of Reckoning still does not damage mobs. It is working on rev https://github.com/TrinityCore/TrinityCore/commit/151ed6f684b47053192d75ea30b0b3b9838b6120

**Comment by illfated** [comment]
Any chance that the suggestion from @tarwyn in comment https://github.com/TrinityCore/TrinityCore/issues/22353#issuecomment-428093704 is useful in this case?
```cpp
    if (effectHandleMode == SPELL_EFFECT_HANDLE_HIT)
    {
        if (!unitCaster)
            return;

        // Hand of Reckoning
        if (m_spellInfo->Id == 62124)
        {
            if (!m_targets.GetUnitTarget() || m_targets.GetUnitTarget()->GetTarget() != unitCaster->GetGUID())
                unitCaster->CastSpell(nullptr, 67485, true);
        }

        return;
    }
```
... or is this just a hackfix like so much else presented from random users?

**Comment by Mysterium** [comment]
Weird Info: with .gm on it works!

**Comment by illfated** [comment]
Interesting feedback. Would be nice if any of the regular coders could test & verify that, then see if GM characters skip any checks which regular characters are unable to pass.

**Comment by lachtanek** [comment]
probably because creatures can't have GMs in threat list

**Comment by illfated** [comment]
Good point. So there is something wrong in how threat is handled in relation to HoR,  I guess.

**Comment by lachtanek** [comment]
my guess would be that the player gets aggro before SpellScript runs and then it runs and sees that the mob is already attacking the player so it does nothing

**Comment by Langerz82** [comment]
Fix: Please test.


```diff
diff --git a/src/server/game/Spells/SpellEffects.cpp b/src/server/game/Spells/SpellEffects.cpp
index 42c54b0fa4..6eaf46ec89 100644
--- a/src/server/game/Spells/SpellEffects.cpp
+++ b/src/server/game/Spells/SpellEffects.cpp
@@ -3078,8 +3078,12 @@ void Spell::EffectTaunt(SpellEffIndex /*effIndex*/)
     ThreatManager& mgr = unitTarget->GetThreatManager();
     if (mgr.GetCurrentVictim() == unitCaster)
     {
-        SendCastResult(SPELL_FAILED_DONT_REPORT);
-        return;
+        Creature* creature = unitTarget->ToCreature();
+        if (creature && creature->GetReactState() != REACT_PASSIVE)
+        {
+            SendCastResult(SPELL_FAILED_DONT_REPORT);
+            return;
+        }
     }
 
     // Hand of Reckoning
```

**Comment by D0d0** [comment]
> Any chance that the suggestion from @tarwyn in comment [#22353 (comment)](https://github.com/TrinityCore/TrinityCore/issues/22353#issuecomment-428093704) is useful in this case?
> 
> ```c++
>     if (effectHandleMode == SPELL_EFFECT_HANDLE_HIT)
>     {
>         if (!unitCaster)
>             return;
> 
>         // Hand of Reckoning
>         if (m_spellInfo->Id == 62124)
>         {
>             if (!m_targets.GetUnitTarget() || m_targets.GetUnitTarget()->GetTarget() != unitCaster->GetGUID())
>                 unitCaster->CastSpell(nullptr, 67485, true);
>         }
> 
>         return;
>     }
> ```
> 
> ... or is this just a hackfix like so much else presented from random users?

Do not forget, I should not do damage to players. https://www.ownedcore.com/forums/world-of-warcraft/world-of-warcraft-exploits/257608-hand-of-reckoning-usage-pvp.html

**Comment by Wyrserth** [comment]
Could not reproduce this issue, tried with [one fleeing mob spawned in GM Island](https://imgur.com/a/dZyLeLx).

On the other hand, #9068 is still bugged. Doesn't deal damage to player pets.

**Comment by Killyana** [comment]
I tested with npc=430 and when the creature is  fleeing it doesn't take damage from Hand of Reckoning.

**Comment by Wyrserth** [comment]
It works for me [even for that creature](https://imgur.com/a/jdpo2Lq), but there's another issue: if it's casting before fleeing, it takes a bit for it to drop the target and start fleeing. Damage can only be dealt after it drops the target.

I'm not sure if that's blizzlike.

**Comment by Killyana** [comment]
Weird, for me it's impossible to deal damage whatever the creature having my char in target or not.

**Comment by Wyrserth** [comment]
Fixed in 5dc2dfd60c4ecd243729e86496d926976e383156.
